src/Keboola/DbExtractor/Application.php

```diff
@@ -4,73 +4,89 @@
 
 namespace Keboola\DbExtractor;
 
-use Keboola\Component\Logger;
+use Keboola\Component\BaseComponent;
 use Keboola\DbExtractor\Configuration\ActionConfigDefinition;
+use Keboola\DbExtractor\Configuration\BaseExtractorConfig;
 use Keboola\DbExtractor\Configuration\ConfigDefinition;
 use Keboola\DbExtractor\Configuration\ConfigRowDefinition;
-use Keboola\DbExtractor\Exception\UserException;
+use Keboola\Component\UserException;
 use Keboola\DbExtractor\Extractor\Extractor;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Config\Definition\ConfigurationInterface;
 use Symfony\Component\Config\Definition\Exception\Exception as ConfigException;
-use Symfony\Component\Config\Definition\Processor;
-use ErrorException;
 
-class Application
+class Application extends BaseComponent
 {
-    /** @var string */
-    private $action;
-
-    /** @var array */
-    private $parameters;
-
-    /** @var array */
-    private $state;
-
-    /** @var Logger */
-    private $logger;
-
     /** @var Extractor */
     private $extractor;
 
     /** @var ConfigurationInterface */
     private $configDefinition;
 
-    public function __construct(array $config, Logger $logger, array $state = [])
-    {
-        static::setEnvironment();
+    /** @var string */
+    private $configDefinitionClassName;
+
+    public function __construct(
+        array $config,
+        LoggerInterface $logger,
+        array $state = [],
+        ?ConfigurationInterface $customConfigDefinition = null
+    ) {
+        if ($customConfigDefinition) {
+            $this->configDefinition = $customConfigDefinition;
+        }
 
-        $this->action = $config['action'];
-        $this->parameters = $config['parameters'];
-        $this->state = $state;
-        $this->logger = $logger;
+        $this->setConfigDefinitionClassNameByConfig($config);
 
-        $extractorFactory = new ExtractorFactory($this->parameters, $this->state);
-        $this->extractor = $extractorFactory->create($this->logger);
+        parent::__construct($logger);
 
-        if ($this->action !== 'run') {
-            $this->configDefinition = new ActionConfigDefinition();
-        } elseif (isset($this->parameters['tables'])) {
-            $this->configDefinition = new ConfigDefinition();
+        $extractorFactory = new ExtractorFactory($this->getConfig()->getParameters(), $state);
```

Comment:
neslo by to resit pres (abstraktni) metodu `getExtractorClass` ? - stejne tak je `getConfigClass` a odpadne tim cela ta habadura s pridavanim extractor_class do parametru

Reply:
Slo, ale je to k nicemu, cela $extractorFactory hned v dalsim PR vyzmizi.

Reply:
ok, hlavni je, aby sla pryc